yt_dlp/extractor/agora.py

```diff
@@ -165,18 +165,16 @@ def _real_extract(self, url):
         metadata = metadata[0]
 
         formats = []
-        for ext in ('aac', 'mp3'):
-            url_data = self._download_json(
-                f'https://api.podcast.radioagora.pl/api4/getSongUrl?podcast_id={media_id}&device_id={uuid.uuid4()}&ppre=false&audio={ext}',
-                media_id, f'Downloading podcast {ext} URL')
-            # prevents inserting the mp3 (default) multiple times
-            if 'link_ssl' in url_data and f'.{ext}' in url_data['link_ssl']:
-                formats.append({
-                    'url': url_data['link_ssl'],
-                    'ext': ext,
-                    'vcodec': 'none',
-                    'acodec': ext,
-                })
+        url_data = self._download_json(
+            f'https://api.podcast.radioagora.pl/api4/getSongUrl?podcast_id={media_id}&device_id={uuid.uuid4()}&ppre=false&audio=mp3',
+            media_id, 'Downloading podcast mp3 URL')
+        if 'link_ssl' in url_data:
+            formats.append({
+                'url': url_data['link_ssl'],
+                'ext': 'mp3',
+                'vcodec': 'none',
+                'acodec': 'mp3',
+            })
```

Comment:
I would even further simplify this. Remove the `formats` var init above, inline the `url`/`ext`/codec fields in the return (`ext` is redundant now, core code will set it from `acodec`)

```suggestion
        mp3_url = self._download_json(
            'https://api.podcast.radioagora.pl/api4/getSongUrl',
            media_id, 'Downloading podcast mp3 URL', query={
                'podcast_id': media_id,
                'device_id': str(uuid.uuid4()),
                'ppre': 'false',
                'audio': 'mp3',
            })['link_ssl']
```
```diff
         return {
             'id': media_id,
-            'formats': formats,
+            'url': mp3_url,
+            'vcodec': 'none',
+            'ext': 'mp3',
```